### PR TITLE
few improvements to tracking DQM

### PR DESCRIPTION
**Pull request by VinInn**
removed 2D histos for candidate
added histos for Pixel layers
make provision to monitor forward tracks

### CODE REVIEW

**Comment by VinInn** [comment]
@cmsbuild, please test

**Comment by VinInn** [comment]
@cmsbuild, please test

File: DQM/TrackingMonitor/src/TrackBuildingAnalyzer.cc [hunk 1]
```diff
@@ -266,13 +266,13 @@ void TrackBuildingAnalyzer::initHisto(DQMStore::IBooker & ibooker, const edm::Pa
     numberOfTrajCandsPerSeedVsEta->setAxisTitle("Seed #eta", 1);
     numberOfTrajCandsPerSeedVsEta->setAxisTitle("Number of Trajectory Candidates for each Seed", 2);
 
-    histname = "SeedStoppingSourceVsNumberOfTrajCandsPerSeed_"+seedProducer.label() + "_";
-    seedStoppingSourceVsNumberOfTrajCandsPerSeed = ibooker.book2D(histname+CatagoryName, histname+CatagoryName, candsBin, candsMin, candsMax, stopReasonSize, 0, stopReasonSize);
+    histname = "StoppingSourceVsNumberOfTrajCandsPerSeed_"+seedProducer.label() + "_";
```
Comment (makortel) [inline DQM/TrackingMonitor/src/TrackBuildingAnalyzer.cc]: I'd suggest to keep the "Seed" prefix to highlight the difference wrt. the "track stopping source histograms" and instead fix a likely bug caused by #20450 by renaming
* line 244: "StoppingSourceVsPhi_" -> "SeedStoppingSourceVsPhi_"
* line 249: "StoppingSourceVsEta_" -> "SeedStoppingSourceVsEta_"

(or, at least the histogram naming bug should be fixed, I don't really mind if this histogram starts with "Seed" or not as long as it is clear that it is the "seed stopping reason")

Comment (VinInn) [reply DQM/TrackingMonitor/src/TrackBuildingAnalyzer.cc]: ok, I will have a look, I was trying to keep histogram together as they are plotted in alphabetic order...

File: DQMOffline/RecoB/python/PrimaryVertexMonitor_cff.py [hunk 2]
```diff
@@ -25,3 +25,10 @@
    EtaMax         = cms.double(2.5),
    EtaMin         = cms.double(-2.5)
 )
+
+# same as above, should be in sync with cut used in Vertex finder...
+from Configuration.Eras.Modifier_phase1Pixel_cff import phase1Pixel
+from Configuration.Eras.Modifier_phase2_tracker_cff import phase2_tracker
+phase1Pixel.toModify(pvMonitor, EtaBin=26, EtaMin=-2.5, EtaMax=2.5)
+phase2_tracker.toModify(pvMonitor, EtaBin=26, EtaMin=-2.5, EtaMax=2.5)
```
Comment (makortel) [inline DQMOffline/RecoB/python/PrimaryVertexMonitor_cff.py]: To my understanding phase2 has eta cut in 4
https://github.com/cms-sw/cmssw/blob/71a61e3cb24ecf0a7647c6e1a1b6871ccf5e4e74/RecoVertex/PrimaryVertexProducer/python/OfflinePrimaryVertices_cfi.py#L57-L60

File: DQM/TrackingMonitor/src/TrackBuildingAnalyzer.cc (hunk 1 above)
Comment (makortel) [reply DQM/TrackingMonitor/src/TrackBuildingAnalyzer.cc]: Ok, maybe move the "Seed" to some other place in the name or something then?

**Comment by VinInn** [comment]
@cmsbuild, please test

**Comment by VinInn** [comment]
do not see any error in
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21357/24534/runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/

and step5 strange...

**Comment by VinInn** [comment]
@cmsbuild, please test

**Comment by VinInn** [comment]
Is there any reason why this PR is not yet merged?

**Comment by davidlange6** [comment]
its not signed by dqm

**Comment by VinInn** [comment]
is there any show stopper for DQM not signing?

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1
i guess not

**Pull request merged**